changes/ce/fix-13216.en.md

```diff
@@ -0,0 +1,10 @@
+Respcet `clientid_prefix` config for MQTT bridges.
+
+As of version 5.4.1, EMQX limits MQTT Client ID lengths to 23 bytes.
+Previously, the system included the `clientid_prefix` in the hash calculation of the original, excessively long Client ID, thereby impacting the resulting shortened ID.
+
+Change Details:
+- Without Prefix: Behavior remains unchanged; EMQX will hash the entire Client ID into a 23-byte space (when longer than 23 bytes).
+- With Prefix:
+  - Prefix no more than 19 bytes: The prefix is preserved, and the remaining suffix is hashed into a 4-byte space.
```

Comment:
Nit. Isn't it 3 bytes in this case, with 4th being a colon?

Reply:
colon has been removed.

Reply:
Ah right, sorry. Then there's an outdated comment that mentions 3 bytes + a colon?

Reply:
i.e. if there is a delimiter required, it should be included in the prefix.
e.g. in the reported issue it's like this: `<org>-<15-char-orgnum>-<random>`,
so the prefix should be configured like `org-abc123-`

Reply:
Oh I see now, thanks for clarifying.